protostar/configuration_file/configuration_file_v1.py

```diff
@@ -27,25 +27,24 @@ class ConfigurationFileV1Model:
     profile_name_to_shared_command_config: dict[ProfileName, CommandConfig]
 
 
+class CommandNamesProvider(Protocol):
```

Comment:
I would make it a class, not protocol (Double inheritance is hard to reason about)

Reply:
The class would have to know how to retrieve data from `ProtostarCLI`. It creates a cyclic dependency between `protostar.cli` and `protostar.configuration_file`. 

> (Double inheritance is hard to reason about)

In my mind:
`class CommandNamesProviderDelegate(Protocol)` == `interface CommandNamesProviderDelegate`

Reply:
Ok let's leave it like this, but please do not rename it to Delegate, you can consider `Interface` or leaving it as it is